### ISSUE TITLE
Add support for generating arbitrary Extension arrays to the fuzzer

### ISSUE THREAD

**Issue by robert3005** [vortex-data/vortex#1588]

**Comment by connortsui20** [comment]
I don't think this is crazy difficult to add with a little help.

If you would like to contribute, please let us know if you are interested in this!

**Comment by robert3005** [comment]
The difficult part is having to know what’s the valid domain of storage type for the extension type. Requires extension types to have validation function for a value

**Comment by connortsui20** [comment]
I think the fuzzer implementation of the extensions arrays can easily "plug in" what a valid domain is and then the fuzzer can always check that the values stay in the domain

**Comment by Pratham1812** [comment]
Hey I would like to pick this up.

**Comment by Pratham1812** [comment]
Hi I have started working on this issue, while coding I found out there are places where the the code for extension arrays has not been added like for instance here-
`fuzz/src/array/sort.rs`
Can I get some insights? should I raise a draft PR. Like currently there are many instances of unreachable code which states that fuzzing is not supported for extension arrays.

**Comment by connortsui20** [comment]
@Pratham1812 I would classify those things as part of this issue: implement as much as possible so that we can add extension types to the fuzzer in a meaningful way. Note that does not mean implement every possible thing (for example, we do not implement sort for the list types in the fuzzer or as compute functions), but we do need that for the other "primitive" types.

**Comment by connortsui20** [comment]
I'll also mention that I set this as a good first issue not because it is a quick or easy change, but because it introduces you to a good percentage of the codebase and requires adding a small amount of code to lots of places (essentially a lot of pattern matching). So please don't hesitate to ask questions!

_Also, a lot of us are on holiday until next year so we might only be able to help you next year_

**Comment by Pratham1812** [comment]
Yeah this issue made me dive deep into the code. Would ask more questions if I face issues. I will try not to disturb you guys during the holidays.

**Comment by connortsui20** [comment]
@Pratham1812 feel free to ask questions! This project is open source, so it's not like we require people to ask questions during specific hours 😅.

Also, if you find something is challenging or difficult to understand, it would be really helpful if you tell us because it is likely something we can improve (in terms of documentation or code organization). Thanks for taking a look at this!
